app/models/supplier.rb

```diff
@@ -5,8 +5,22 @@ class Supplier < ApplicationRecord
   validates :name, presence: true
   validates :address, presence: true
   validates :phone_number, presence: true, length: { in: 8..12, allow_blank: true }
+  validates :opens_at, numericality: { in: 0..23 }
+  validates :close_at, numericality: { in: 0..23 }
 
   def delivery?
     delivery
   end
+
+  def self.categories(category)
+    joins(services: [:categories]).where({ categories: { id: category.id } }).group(:name,:id)
+  end
+
+  def self.search(search)
+    if search
+      where('suppliers.name LIKE ?', "%#{search}%")
+    else
+      all
+    end
```

Comment:
those two methods could be a scope, right? Was there any preference for doing them like class methods?

Reply:
True, nope there was not any preference. I ve modified them into scope in 7ac9b88

Reply:
Okey, i forgot to add grouping parameter,  here it is: 546a385